lib/LogicalExpression.h

```diff
@@ -99,6 +99,123 @@ namespace LogicalExpressionDetail
 		}
 	};
 
+	template <typename ContainedClass>
+	class FalsifiabilityVisitor;
+
+	/// Visitor to test whether expression's value can be true
+	template <typename ContainedClass>
+	class SatisfiabilityVisitor : public boost::static_visitor<bool>
+	{
+		typedef ExpressionBase<ContainedClass> Base;
+
+		std::function<bool(const typename Base::Value &)> satisfiabilityTest;
+		FalsifiabilityVisitor<ContainedClass> *falsifiabilityVisitor;
```

Comment:
falsifiability[Visitor, Test], satisfiability[Visitor, Test], countSatisfiable, countFalsifiable should be extracted to common base class template.

smth like:

```
template <typename ContainedClass>
	class BaseVisitor : public boost::static_visitor<bool> ...

template <typename ContainedClass>
	class SatisfiabilityVisitor : public BaseVisitorContainedClass>
```